homework/index.js

```diff
@@ -30,15 +24,98 @@
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+  function renderError(error) {
+    const root = document.getElementById('root');
+    createAndAppend('h1', root, { text: error.message });
+  }
+
+  function createOptions(repositoryDetails, select) {
+    repositoryDetails.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function addRow(tbody, labelText, value, str) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${labelText}`, class: str });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoInformation(repository, container) {
+    container.innerHTML = ' ';
+    const table = createAndAppend('table', container, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository: ', '', 'explanations');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
     });
+    addRow(tbody, 'Description:', repository.description, 'explanations');
+    addRow(tbody, 'Fork:', repository.forks, 'explanations');
+    addRow(tbody, 'Updated:', `${repository.updated_at}`.substring(0, 10), 'explanations');
+  }
+
+  async function renderContributorsInformation(repository, container) {
+    try {
+      await fetchJSON(repository.contributors_url).then(contributorDetails => {
+        container.innerHTML = ' ';
+        const div = document.getElementById('cont-ulist');
+        createAndAppend('h4', div, { text: 'Contributors' });
+        const contrList = createAndAppend('ul', div, { class: 'contr-list' });
+        contributorDetails.forEach(contributor => {
+          const li = createAndAppend('li', contrList);
+          const a = createAndAppend('a', li, { href: contributor.html_url, target: '_blank' });
+          createAndAppend('img', a, {
+            src: contributor.avatar_url,
+            class: 'images',
+          });
+          createAndAppend('span', a, {
+            text: contributor.login,
+          });
+          createAndAppend('span', a, {
+            text: contributor.contributions,
+            class: 'contributions',
+          });
+        });
+      });
+    } catch (err) {
+      renderError(err);
+    }
+  }
+
+  async function main(url) {
+    const root = document.getElementById('root');
+    const menuSection = createAndAppend('section', root, { id: 'menu-section' });
+    createAndAppend('p', menuSection, { text: 'HYF Repositories', id: 'hyf-text' });
+    const select = createAndAppend('select', menuSection, { id: 'select-button' });
+    const bodyDiv = createAndAppend('div', root, { id: 'body-div' });
+    const section = createAndAppend('section', bodyDiv, { class: 'repo-info-list' });
+    const contrDiv = createAndAppend('div', bodyDiv, {
```

Comment:
The semantics of your HTML tags seems somewhat disorganized. You have `section` for the menu, a `div` for the `body-div`, then as its children, another `section` and a `div`:

```html
<div id="root">
  <section id="menu-section"><!-- ... --></section>
  <div id="body-div">
    <section class="repo-info-list"><!-- ... --></section>
    <div class="contributors-div" id="cont-ulist"><!-- ... --></div>
  </div>
</div>
```

Instead, consider this:

```html
<div id="root">
  <header id="menu-section"
    <!-- ... -->
  </header>
  <main id="body-div">
    <section class="repo-info-list">
      <!-- ... -->
    </section>
    <section class="contributors-div" id="cont-ulist">
      <!-- ... -->
    </section>
  </main>
</div>
```